File: C++/C++Basic/06.Conditional-Statements-Advanced-Exercise/03.newHouse.cpp
#include <iostream>

using namespace std;

int main()
{
    string typeFlowers;
    int countFlowers, budget;

    cin >> typeFlowers;
    cin >> countFlowers >> budget;

    double price = 0.0;

    if (typeFlowers == "Roses")
    {
        price = 5.0;
    }
    else if (typeFlowers == "Dahlias")
    {
        price = 3.80;
    }
    else if (typeFlowers == "Tulips")
    {
        price = 2.80;
    }

    else if (typeFlowers == "Narcissus")
    {
        price = 3.0;
    }
    else if (typeFlowers == "Gladiolus")
    {
        price = 2.50;
    }

    double totalPrice = price * countFlowers;

    if (countFlowers > 80 && typeFlowers == "Tulips"){
        totalPrice *= 0.85;
    }
    else if (countFlowers < 120 && typeFlowers == "Narcissus")
    {
        totalPrice *= 1.15;
    }
    else if (countFlowers < 80 && typeFlowers == "Gladiolus"){
        totalPrice *= 1.20;
    }
    else if (countFlowers > 90 && typeFlowers == "Dahlias")
    {
        totalPrice *= 0.85;
    }
    else if (countFlowers > 80 && typeFlowers == "Roses")
    {
        totalPrice *= 0.90;
    }


    cout.setf(ios::fixed);
    cout.precision(2);

    if (budget >= totalPrice){
        
        double leftMoney = budget - totalPrice;
        cout << "Hey, you have a great garden with " << countFlowers << " "<< typeFlowers << " and " << leftMoney <<" leva left." << endl;
    }
    else
    {
        double moneyNeeded = totalPrice - budget;

        cout << "Not enough money, you need " << moneyNeeded << " leva more." << endl;
    }
    return 0;
}

File: Python/Basic/06.Conditional-Statements-Advanced-Exercise/03.new_house.py
flower_type = input()
number_of_flowers = int(input())
budget = int(input())

price = 0

if flower_type == "Roses":
    price = 5
elif flower_type == "Dahlias":
    price = 3.80
elif flower_type == "Tulips":
    price = 2.80
elif flower_type == "Narcissus":
    price = 3
else:
    price = 2.50

price *= number_of_flowers

if flower_type == "Roses" and number_of_flowers > 80:
    price  *= 0.90
elif flower_type == "Dahlias" and number_of_flowers > 90:
    price *= 0.85
elif flower_type == "Tulips" and number_of_flowers > 80:
    price *= 0.85
elif flower_type == "Narcissus" and number_of_flowers < 120:
    price *= 1.15
elif flower_type == "Gladiolus" and number_of_flowers < 80:
    price *= 1.20

if budget >= price:
    money_left = budget-price
    print(f"Hey, you have a great garden with {number_of_flowers} {flower_type} and {money_left:.2f} leva left.")
else:
    money_needed = price-budget
    print(f"Not enough money, you need {money_needed:.2f} leva more.")
